setup.cfg

```diff
@@ -24,4 +24,4 @@ license_file = LICENSE.txt
 
 [flake8]
 max-line-length = 88
-ignore = W503, E121, E126, E211, E225, E501, E203, E402, F401, F811, E231
+ignore = W503, E121, E126, E211, E225, E501, E203, E402, F401, F811, E231, F901
```

Comment:
why is this needed?

Reply:
In the `RasaModel` we have some methods that raise a `NotImplemented` instead of a `NotImplementedError` to indicate that those method should not be implemented at all. (cc @Ghostvv ).

Reply:
as far as I understand from python docs. For an abstract class, if a method should be implemented in a subclass, it should raise `NotImplementedError`. But there are a number of methods in superclass of our class, that we don't want to be implemented. So I raise `NotImplemented`. I'm not sure if it is the correct approach though

Reply:
ah ok that makes sense, but won't work this way (`NotImplemented ` does not inherit from `Exception` so it can not be raised).

I'd suggest
* undo the above change reinstating the flake test
* change `raise NotImplemented` to `raise Exception("This method should neither be called nor implemented in our code.")` or something like that

Reply:
true, should I `return NotImplemented`? https://docs.python.org/3.7/library/constants.html#NotImplemented

Reply:
I think `NotImplemented` should be used for different cause, I'll raise `Exception` per your suggestion

Reply:
sounds good